### PR TITLE
Make terminals stats section in perfviewEditor

### PR DESCRIPTION
**Pull request by Tyriar**
Part of #185393

image

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 1]
```diff
@@ -293,8 +292,7 @@ export class TerminalService implements ITerminalService {
 		}
 		reconnectedPromise.then(() => {
 			this._setConnected();
-			this._mark('code/terminal/didReconnect');
-			this._timerService.setPerformanceMarks('terminal', this._perfMarks);
+			this._mark('terminal/didReconnect');
```
Comment (jrieken) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts:295]: It's fine to roll your own thing but you aren't included in perf telemetry unless you use `ITimerService#setPerformanceMarks`. So, I would rather do `perf('code/terminal/...')` and later `ITimerService#getPerformanceMarks`

Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:295]: Good to know, I've also got localPtyHost/remotePtyHost raw marks coming which use that to track those procs

**Comment by Tyriar** [comment]
I made the changes, it required a bunch of shuffling around of "terminal backends" to move them into platform so they could be used in the timer service. In the big https://github.com/microsoft/vscode/pull/185754/commits/ed54d72d8c3e068055e57416d42c9cf5fc0921c3 commit the main point of interest is the change in timerService.ts

**Review by jrieken — APPROVED** [review]
perfEditor and timerService changes are looking good (didn't check the other ;-))

**Pull request merged**